### ISSUE TITLE
horizon /paths endpoint returns paths to destination_account for assets that destination_account doesn't have trustlines for

### ISSUE THREAD

**Issue by patrickxb** [stellar/go#1128]
The horizon `/paths` endpoint currently returns paths that have no chance of ever succeeding because the destination_account does not have a trustline to the destination asset.

The `/paths` endpoint has a destination_account parameter that I believe is unused.

I was expecting all of the paths returned by the `/paths` endpoint to succeed if submitted as a path payment operation.

For example:

https://horizon.stellar.org/paths?source_account=GBZX4364PEPQTDICMIQDZ56K4T75QZCR4NBEYKO6PDRJAHZKGUOJPCXB&destination_account=GBA3ZJS4ECOCDF3E3ILAMWGNX25IXORCEYXNPARMCJRCPX7FW3JKGSQT&destination_asset_type=credit_alphanum4&destination_asset_code=USD&destination_asset_issuer=GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX&destination_amount=1

currently returns

```
{
  "_embedded": {
    "records": [
      {
        "source_asset_type": "native",
        "source_amount": "7.9470004",
        "destination_asset_type": "credit_alphanum4",
        "destination_asset_code": "USD",
        "destination_asset_issuer": "GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX",
        "destination_amount": "1.0000000",
        "path": []
      },
      {
        "source_asset_type": "native",
        "source_amount": "8.3433538",
        "destination_asset_type": "credit_alphanum4",
        "destination_asset_code": "USD",
        "destination_asset_issuer": "GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX",
        "destination_amount": "1.0000000",
        "path": [
          {
            "asset_type": "credit_alphanum4",
            "asset_code": "WSD",
            "asset_issuer": "GDSVWEA7XV6M5XNLODVTPCGMAJTNBLZBXOFNQD3BNPNYALEYBNT6CE2V"
          }
        ]
      },
      {
        "source_asset_type": "native",
        "source_amount": "10.1429220",
        "destination_asset_type": "credit_alphanum4",
        "destination_asset_code": "USD",
        "destination_asset_issuer": "GDUKMGUGDZQK6YHYA5Z6AY2G4XDSZPSZ3SW5UN3ARVMO6QSRDWP5YLEX",
        "destination_amount": "1.0000000",
        "path": [
          {
            "asset_type": "credit_alphanum4",
            "asset_code": "EURT",
            "asset_issuer": "GAP5LETOV6YIE62YAM56STDANPRDO7ZFDBGSNHJQIYGGKSMOZAHOOS2S"
          },
          {
            "asset_type": "credit_alphanum4",
            "asset_code": "WSD",
            "asset_issuer": "GDSVWEA7XV6M5XNLODVTPCGMAJTNBLZBXOFNQD3BNPNYALEYBNT6CE2V"
          }
        ]
      }
    ]
  }
}
```

The destination account GBA3ZJS4E... has no trustline to USD/GDUKMGUGDZ... thus submitting any of these path payments results in an error.

**Comment by marcinx** [comment]
I noticed the same. It appears that the `destination_account` parameter is being ignored. You can put any arbitrary account in there and it will return all possible path combinations regardless of any trustlines that exist on the destination account.

In my opinion the `destination_account` should be optional. If given then the query should be constrained by trustlines on the account as outlined by @patrickxb above. If omitted it behave exactly as it does currently and return all possible path combinations. This is also a use case in some situations and we actually use it this way in some places.

**Comment by bartekn** [comment]
We believe this has been fixed in `0.20.0` release (and `0.20.1` that contains some performance fixes). Horizon `0.20.1rc1` is currently deployed to https://horizon.stellar.org. Please check it out. It is using the new in-memory order book graph.

**Comment by bartekn** [comment]
Yes, you need `ENABLE_EXPERIMENTAL_INGESTION=true` env variable set.
